bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/amp/AmpTransformer.java

```diff
@@ -195,7 +195,26 @@ private String getJsContent() {
                 headLibResource = Utils.resolveResource(resolver, headLibPath);
                 if (headLibResource == null) {
                     LOG.trace("No custom headlib for resource type {}.", resourceType);
-                    continue;
+
+                    // Get resource from the resource type.
+                    Resource coreResource = slingRequest.getResourceResolver().getResource(resourceType);
+                    if (coreResource == null) {
+                        LOG.debug("Can't access resource from resource type {}.", resourceType);
+                        continue;
+                    }
+
+                    // Get resource superType path from the resource type.
+                    String superTypePath = coreResource.getResourceSuperType();
```

Comment:
just a single check for supertype? this should be in a loop to check if _any_ super types have one

Reply:
This code would probably work:
```
    private Resource getSuperTypeHeadLib(ResourceResolver resolver, Resource resource) {
        if (resource == null) {
            return null;
        }
        String superType = resource.getResourceSuperType();
        if (StringUtils.isBlank(superType)) {
            return null;
        }
        Resource superTypeHeadlib = Utils.resolveResource(resolver, superType + "/" + cfg.getHeadlibName() + "/" + JcrConstants.JCR_CONTENT);
        if (superTypeHeadlib == null) {
            return getSuperTypeHeadLib(resolver, Utils.resolveResource(resolver, superType));
        }
        return null;
    }

Reply:
Fixed!